dipy/nn/deepn4.py

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/python
+"""
+Class and helper functions for fitting the DeepN4 model.
+"""
+import logging
+import numpy as np
+
+import nibabel as nib
+from scipy.ndimage import gaussian_filter
+from dipy.nn.utils import transform_img, recover_img
+
+from dipy.data import get_fnames
+from dipy.testing.decorators import doctest_skip_parser
+from dipy.utils.optpkg import optional_package
+from dipy.nn.utils import set_logger_level
+
+tf, have_tf, _ = optional_package('tensorflow')
+tfa, have_tfa, _ = optional_package('tensorflow_addons')
+if have_tf and have_tfa:
```

Comment:
Can you change this part like in Synb0? I add the Layer and Model as None because otherwise without tensorflow importing this throws an error.

Reply:
added     ```
from tensorflow_addons.layers import InstanceNormalization
else:
    class Model:
        pass

    class Layer:
        pass
    logging.warning('This model requires Tensorflow and Tensorflow\
```